ci/release/update-version.sh

```diff
@@ -0,0 +1,62 @@
+#!/bin/bash
+########################
+# cuDF Version Updater #
+########################
+
+## Usage
+# bash update-version.sh <type>
+#     where <type> is either `major`, `minor`, `patch`
+
+set -e
+
+# Grab argument for release type
+RELEASE_TYPE=$1
+
+# Get current version and calculate next versions
+CURRENT_TAG=`git describe --abbrev=0 --tags | tr -d 'v'`
+CURRENT_MAJOR=`echo $CURRENT_TAG | awk '{split($0, a, "."); print a[1]}'`
+CURRENT_MINOR=`echo $CURRENT_TAG | awk '{split($0, a, "."); print a[2]}'`
+CURRENT_PATCH=`echo $CURRENT_TAG | awk '{split($0, a, "."); print a[3]}'`
+NEXT_MAJOR=$((CURRENT_MAJOR + 1))
+NEXT_MINOR=$((CURRENT_MINOR + 1))
+NEXT_PATCH=$((CURRENT_PATCH + 1))
+NEXT_FULL_TAG=""
+NEXT_SHORT_TAG=""
+
+# Determine release type
+if [ "$RELEASE_TYPE" == "major" ]; then
+  NEXT_FULL_TAG="${NEXT_MAJOR}.0.0"
+  NEXT_SHORT_TAG="${NEXT_MAJOR}.0"
+elif [ "$RELEASE_TYPE" == "minor" ]; then
+  NEXT_FULL_TAG="${CURRENT_MAJOR}.${NEXT_MINOR}.0"
+  NEXT_SHORT_TAG="${CURRENT_MAJOR}.${NEXT_MINOR}"
+elif [ "$RELEASE_TYPE" == "patch" ]; then
+  NEXT_FULL_TAG="${CURRENT_MAJOR}.${CURRENT_MINOR}.${NEXT_PATCH}"
+  NEXT_SHORT_TAG="${CURRENT_MAJOR}.${NEXT_MINOR}"
+else
+  echo "Incorrect release type; use 'major', 'minor', or 'patch' as an argument"
+  exit 1
+fi
+
+# Move to root of repo
+cd ../..
+echo "Preparing '$RELEASE_TYPE' release [$CURRENT_TAG -> $NEXT_FULL_TAG]"
+
+# Inplace sed replace; workaround for Linux and Mac
+function sed_runner() {
+    sed -i.bak ''"$1"'' $2 && rm -f ${2}.bak
+}
+
+# cpp update
+sed_runner 's/'"CUDA_DATAFRAME VERSION ${CURRENT_TAG}"'/'"CUDA_DATAFRAME VERSION ${NEXT_FULL_TAG}"'/g' cpp/CMakeLists.txt
+
+# libcudf
+sed_runner 's/version=.*/version=\"'"${NEXT_FULL_TAG}"'\",/g' cpp/python/setup.py
+
+# Conda recipe updates
+sed_runner 's/libcudf .*/libcudf '"${NEXT_SHORT_TAG}*"'/g' conda/recipes/cudf/meta.yaml
+sed_runner 's/libcudf_cffi .*/libcudf_cffi '"${NEXT_SHORT_TAG}*"'/g' conda/recipes/cudf/meta.yaml
+
+# RTD update
+sed_runner 's/version = .*/version = '"'${NEXT_SHORT_TAG}'"'/g' docs/source/conf.py
+sed_runner 's/release = .*/release = '"'${NEXT_FULL_TAG}'"'/g' docs/source/conf.py
```

Comment:
These sed_runner commands should be changed to suit this project.

For example in your `/conda-recipes/meta.yml` file, you can change the cudf version to the latest stable version using this script. Any lines in your code that rely on specific versions of things (especially cudf) should be utilizing this.